src/QMCWaveFunctions/detail/CUDA/cuBLAS_LU.cu

```diff
@@ -0,0 +1,212 @@
+    return cudaSuccess;
+
+  const int COLBS          = 256;
+  const int num_col_blocks = (n + COLBS - 1) / COLBS;
+  dim3 dimBlock(COLBS);
+  dim3 dimGrid(batch_size, num_col_blocks);
+  computeLogDet_kernel<COLBS>
+      <<<dimGrid, dimBlock, 0, hstream>>>(n, lda, CUDATYPECAST(LU_mat), pivots, CUDATYPECAST(logdets));
+  return cudaPeekAtLastError();
+}
+
+template<typename T>
+void computeLogDet_batched(cudaStream_t& hstream,
+                           const int n,
+                           const int lda,
+                           T** LU_mat,
+                           const int* pivots,
+                           std::complex<double>* log_dets,
+                           const int batch_size)
+{
+  cudaErrorCheck(computeLogDet_batched_impl(hstream, n, lda, LU_mat, pivots, log_dets, batch_size),
+                 "fused kernel call failed");
+}
+
+/** Takes the transpose of PsiM using LU factorization calculates the log determinant and invPsiM
+ *
+ *  \param[inout] Ms -       pointers to pointers to working memory for Ms that are used to return invMs
+ *  \param[in]    pivots -   pointer to n * nw ints allocated in device memory for pivots array.
+ *  \param[in]    infos -    pointer to nw ints allocated in device memory factorization infos
+ *  \param[out]   log_dets - pointer device memory for nw log determinant values to be returned, 
+ *                           maybe this is supposed to be just RealType
+ *  \param[in]    batch_size - if this changes over run a huge performance hit will be taken as memory allocation syncs device.
+ */
+void computeInverseAndDetLog_batched(cublasHandle_t& h_cublas,
+                                     cudaStream_t& hstream,
+                                     const int n,
+                                     const int lda,
+                                     double* Ms[],
+                                     double* Cs[],
+                                     double* LU_diags,
+                                     int* pivots,
+                                     int* infos,
+                                     std::complex<double>* log_dets,
+                                     const int batch_size)
+{
+  //LU is returned in Ms
+  cublasErrorCheck(cuBLAS::getrf_batched(h_cublas, n, Ms, lda, pivots, infos, batch_size),
+                   "cuBLAS::getrf_batched failed in computeInverseAndDetLog_batched");
+  // cudaErrorCheck(computeLUDiag_batched_impl(hstream, n, lda, Ms, LU_diags, batch_size),
+  //                "failed to extract LU diag values at cuomputeLUDiag_batched_impl");
+  cudaErrorCheck(computeLogDet_batched_impl(hstream, n, lda, Ms, pivots, log_dets, batch_size),
+                 "failed to calculate log determinant values in computeLogDet_batched_impl");
+  cublasErrorCheck(cuBLAS::getri_batched(h_cublas, n, Ms, lda, pivots, Cs, lda, infos, batch_size),
+                   "cuBLAS::getri_batched failed in computeInverseAndDetLog_batched");
```

Comment:
There would ideally be no transfers at this level at all since this is the level equivalent to the cuBLAS calls.   Transfer and synchronization occurs at the DiracMatrix level.

Reply:
cuBLAS::getri_batched is at the cuBLAS call level. computeInverseAndDetLog_batched is already high level. We must transfer info and check error before exiting. Since anyway, transfer is needed, so prefer to transfer log_dets as well.

Reply:
my understanding is if getrf has no infos getri won't either.